lib/formats/mgh.cpp

```diff
@@ -112,6 +122,7 @@ namespace MR
       for (const auto& tag : MGHO.tags) {
         out.write (reinterpret_cast<const char*> (&tag.id), sizeof(tag.id));
         out.write (reinterpret_cast<const char*> (&tag.size), sizeof(tag.size));
+        assert (size_t(ByteOrder::BE (tag.size)) == tag.content.size());
```

Comment:
That specific case is a write, so it's only testing that MRtrix has set the tag.size value appropriately. Nothing to do with input data.

I'm pretty sure that FreeSurfer does *not* include null terminators (at least not always) in the length calculation: the phase encode direction tag gave a size of 7 for the string `UNKNOWN`.